### PR TITLE
[BACKLOG-38705] - Adding kafka plugin enhancement changes in 10.0

### PR DESCRIPTION
**Pull request by lsaikrishna89**

### CODE REVIEW

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 5m 19s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl kettle-plugins/kafka
```
---
:ok_hand: All tests passed!
 
Tests run: **53**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/big-data-plugin/job/PR-2476/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: kettle-plugins/kafka/src/main/java/org/pentaho/big/data/kettle/plugins/kafka/KafkaConsumerInput.java [hunk 1]
```diff
@@ -59,38 +66,48 @@ public KafkaConsumerInput( StepMeta stepMeta, StepDataInterface stepDataInterfac
...
+
+  protected void prepareConsumer( KafkaConsumerInputMeta kafkaConsumerInputMeta,
+                                  KafkaConsumerInputData kafkaConsumerInputData ) {
+    Consumer consumer = kafkaConsumerInputMeta.getKafkaFactory().consumer( kafkaConsumerInputMeta,
+            this::environmentSubstitute, kafkaConsumerInputMeta.getKeyField().getOutputType(),
+            kafkaConsumerInputMeta.getMessageField().getOutputType() );
 
     Set<String> topics =
-      kafkaConsumerInputMeta.getTopics().stream().map( this::environmentSubstitute ).collect( Collectors.toSet() );
+            kafkaConsumerInputMeta.getTopics().stream().map( this::environmentSubstitute ).collect( Collectors.toSet() );
```
Comment (peterrinehart) [inline kettle-plugins/kafka/src/main/java/org/pentaho/big/data/kettle/plugins/kafka/KafkaConsumerInput.java:104]: Please check your code style settings and fix this and other whitespace changes in this PR.

**Review by peterrinehart — CHANGES_REQUESTED** [review]

**Comment by buildguy** [comment]
### :x: Build failed in 2m 23s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl kettle-plugins/kafka
```
---
:ok_hand: All tests passed!
 
Tests run: **53**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/big-data-plugin/job/PR-2476/2/testReport)

 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 4s
Build command:
```console
mvn clean verify -B -e -Daudit -amd
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 4m 11s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl kettle-plugins/kafka
```
---
:ok_hand: All tests passed!
 
Tests run: **53**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/big-data-plugin/job/PR-2476/4/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 4m 18s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl kettle-plugins/kafka
```
---
:ok_hand: All tests passed!
 
Tests run: **53**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/big-data-plugin/job/PR-2476/5/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 3m 32s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl kettle-plugins/kafka
```
---
:ok_hand: All tests passed!
 
Tests run: **53**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/big-data-plugin/job/PR-2476/6/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 3m 42s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl kettle-plugins/kafka
```
---
:ok_hand: All tests passed!
 
Tests run: **53**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/big-data-plugin/job/PR-2476/7/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Pull request merged**